src/js/data-mapper.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2015 SugarCRM Inc. Licensed by SugarCRM under the Apache 2.0 license.
+ */
+(function() {
+
+    var BDT = window.BDT;
+
+    BDT.dataMap = {
+        // Maps our vardefs fields with the faker.js function to use to generate
+        // a value.
+        fieldNames: {
+            first_name: 'name.firstName',
+            last_name: 'name.lastName',
+            salutation: 'name.prefix',
+            title: 'name.jobTitle',
+            name: 'name.findName',
+
+            billing_address_city: 'address.city',
+            billing_address_country: 'address.country',
+            billing_address_postalcode: 'address.zipCode',
+            billing_address_state: 'address.state',
+            billing_address_street: 'address.streetAddress',
+            primary_address_city: 'address.city',
+            primary_address_country: 'address.country',
+            primary_address_postalcode: 'address.zipCode',
+            primary_address_state: 'address.state',
+            primary_address_street: 'address.streetAddress',
+            alt_address_street: 'address.streetAddress',
+            shipping_address_city: 'address.city',
+            shipping_address_country: 'address.country',
+            shipping_address_postalcode: 'address.zipCode',
+            shipping_address_state: 'address.state',
+            shipping_address_street: 'address.streetAddress',
+            department: 'lorem.word',
+            date_end: 'date.future',
+            date_start: 'date.future',
+            exp_date: 'date.future',
+            repeat_until: 'date.future',
+
+            facebook: 'internet.userName',
+            twitter: 'internet.userName',
+            googleplus: 'internet.userName',
+
+            description: 'lorem.paragraph'
+        },
+        // Maps our field types to the faker.js function to use to generate
+        // a value. If the field to populate is not in the `fieldNames` list above,
+        // we will fallback to this mapping.
+        fieldTypes: {
+            phone: 'phone.phoneNumberFormat',
+            datetime: 'date.recent',
+            date: 'date.past',
+            text: 'lorem.words',
+            bool: 'random.boolean',
+            int: 'random.number',
+            url: 'internet.url',
+            //email: 'internet.exampleEmail'
```

Comment:
we can't add the email type since the `email` fields builds an array, and we can't do that with faker.

Anyway, I think in the future we need to extend faker to be able to provide our own random logic for some of our field. (like building and array of emails)